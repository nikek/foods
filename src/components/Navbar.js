import React, { Component } from 'react'
import { connect } from 'cerebral/react'
import { state, signal } from 'cerebral/tags'

export default connect(
  {
    currentPage: state`app.currentPage`,
    user: state`user`,
    userSettingsRouted: signal`app.userSettingsRouted`,
    homeRouted: signal`app.homeRouted`
  },
  class Navbar extends Component {
    render() {
      const p = this.props

      return (
        <nav className="navbar navbar-dark bg-dark">
          <button
            className="btn btn-primary"
            type="button"
            onClick={p.homeRouted}
          >
            Home
          </button>
          <div className="navbar-brand">Foods</div>
          <img
            src={p.user.photoUrl}
            title={p.user.name}
            alt=""
            width="33"
            height="33"
            className="rounded-circle border border-light"
            onClick={p.userSettingsRouted}
          />
        </nav>
      )
    }
  }
)
