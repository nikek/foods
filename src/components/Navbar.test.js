import React from 'react'
import ReactDOM from 'react-dom'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import { StateContainer } from 'cerebral/react'
import Navbar from './Navbar'

it('renders without crashing', () => {
  const state = {
    user: { name: 'Sherlock', photoUrl: 'http://path-to-img.url' }
  }
  const wrapper = mount(
    <StateContainer state={state}>
      <Navbar />
    </StateContainer>
  )

  expect(wrapper.find('img').length).toBe(1)
  expect(toJson(wrapper)).toMatchSnapshot()
})
