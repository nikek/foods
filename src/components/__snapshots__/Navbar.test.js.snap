// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<StateContainer
  state={
    Object {
      "user": Object {
        "name": "Sherlock",
        "photoUrl": "http://path-to-img.url",
      },
    }
  }
>
  <CerebralWrapping_Navbar>
    <Navbar
      homeRouted={[Function]}
      user={
        Object {
          "name": "Sherlock",
          "photoUrl": "http://path-to-img.url",
        }
      }
      userSettingsRouted={[Function]}
    >
      <nav
        className="navbar navbar-dark bg-dark"
      >
        <button
          className="btn btn-primary"
          onClick={[Function]}
          type="button"
        >
          Home
        </button>
        <div
          className="navbar-brand"
        >
          Foods
        </div>
        <img
          alt=""
          className="rounded-circle border border-light"
          height="33"
          onClick={[Function]}
          src="http://path-to-img.url"
          title="Sherlock"
          width="33"
        />
      </nav>
    </Navbar>
  </CerebralWrapping_Navbar>
</StateContainer>
`;
